test/hw3/test_task3.py

```diff
@@ -0,0 +1,36 @@
+from All_home_works.hw3.task3 import Filter, make_filter, sample_data
+
+
+def test_filter():
+    expected = list(range(2, 100, 2))
+    positive_even = Filter(
+        lambda a: a % 2 == 0, lambda a: a > 0, lambda a: isinstance(a, int)
+    )
+    result = positive_even.apply(range(99))
+    assert expected == result
+
+
+def test_filter_case__data_with_negative():
+    expected = list(range(2, 100, 2))
+    positive_even = Filter(
+        lambda a: a % 2 == 0, lambda a: a > 0, lambda a: isinstance(a, int)
+    )
+    result = positive_even.apply(range(-99, 99))
+    assert expected == result
+
+
+def test_filter_make_functions_valid_for_second():
```

Comment:
Why for a second? Why not for a minute?

Reply:
Rename: test_filter__make_functions__valid_for_only_one_dict